File: src/main/java/com/morak/performancetracker/aop/web/WebConfiguration.java
package com.morak.performancetracker.aop.web;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebConfiguration implements WebMvcConfigurer {

    private final WebMonitor webMonitor;

    public WebConfiguration(WebMonitor webMonitor) {
        this.webMonitor = webMonitor;
    }

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(new PerformanceInterceptor(webMonitor));
    }
}

File: src/main/kotlin/com/morak/performancetracker/aop/web/WebConfiguration.kt
package com.morak.performancetracker.aop.web

import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfiguration(private val interceptor: PerformanceInterceptor) : WebMvcConfigurer {
    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(interceptor)
    }
}
